src/Spark.Engine/Service/Transfer.cs

```diff
@@ -16,11 +17,13 @@ public class Transfer : ITransfer
     {
         ILocalhost localhost;
         IGenerator generator;
+        SparkSettings sparkSettings;
 
-        public Transfer(IGenerator generator, ILocalhost localhost)
+        public Transfer(IGenerator generator, ILocalhost localhost, SparkSettings sparkSettings)
```

Comment:
Since this is a public class default SparkSettings to null so we don't break the signature and handle null value on line 50 and 57

Reply:
done